Gulpfile.js

```diff
@@ -15,9 +17,20 @@ gulp.task('css', function () {
       .pipe(gulp.dest('assets'));
 });
 
-gulp.task('default', ['css', 'connect', 'watch']);
+gulp.task('default', ['css', 'js', 'connect', 'watch']);
+
+gulp.task('js', function () {
+  return gulp.src('assets/js/**/*.js')
+      .pipe(jshint())
+      .pipe(concat('script.js'))
+      .pipe(gulp.dest('assets'));
+});
 
 gulp.task('watch', function () {
+  gulp.watch('assets/js/**/*.js', ['js']).on('change', function () {
+    browserSync.reload();
```

Comment:
I'm not sure if you correctly initialized BrowserSync if I look at their [documentation](https://www.browsersync.io/docs/gulp).